### DISCUSSION TITLE
What is the Quarkus support policy ?

### DISCUSSION THREAD

**Discussion by marcwrobel** [quarkusio/quarkus#29161] [Q&A] [answered]
Dear Quarkus team, I am looking for information about the Quarkus support policy (i.e. maintenance policy) in order [to fix](https://github.com/endoflife-date/endoflife.date/issues/1842) what I think is wrong on [the Quarkus page on endoflife.date](https://endoflife.date/quarkus-framework).

From what I can see in the [Security policy](https://github.com/quarkusio/quarkus/blob/main/SECURITY.md), the Quarkus team is maintaining only the last `major.minor` version of Quarkus. Once a new version is out, the previous version is out of support : security fixes may be backported (no guarantee), but no bug fixes will be backported. Am I correct ?

I also saw there are [Red Hat builds of Quarkus](https://access.redhat.com/support/policy/updates/jboss_notes#p_quarkus). What are they ? Are they different from the builds that can be found on Maven Central ?

**Comment by adriens** [comment]
I'm pinging [Guillaume Smet](https://www.linkedin.com/in/gsmet/) : 

> Hi Guillaume, we are currently adopting Quarkus on our stack. Hence I also wanted to be able to manage our technical debt... and then get EOL/roadmap. I started to contribute on this topic on a dedicated discussion https://github.com/quarkusio/quarkus/discussions/29161 for endoflife.date integration

**Comment by maxandersen** [comment, marked as answer]
TIL today endoflife.date - thanks :) 

Let me do my best to answer your questions:

What is [Red Hat builds of Quarkus](https://access.redhat.com/products/quarkus) - that is Red Hat's offering of providing support and maintenance of Quarkus. The codebase is the same but Red Hat has a productization setup that ensures we have a secured pipeline and reproducable builds of Qaurkus and all critical dependencies. That is not possible to publish to maven central and thus are available from https://maven.repository.redhat.com. The quarkus platform in community is called `io.quarkus.platform.*` in RHBQ its `com.quarkus.redhat.platform.*`. You can see which parts of Quarkus has support etc. on https://code.quarkus.redhat.com. The release frequency is though lower than in community, i.e. latest RHBQ is for 2.7.x which is about 6 months old and RHBQ do releases approximately every 6 months.

> From what I can see in the [Security policy](https://github.com/quarkusio/quarkus/blob/main/SECURITY.md), the Quarkus team is maintaining only the last `major.minor` version of Quarkus. Once a new version is out, the previous version is out of support : security fixes may be backported (no guarantee), but no bug fixes will be backported. Am I correct?

That is correct in general for community. We encourage and support to move fast and steadily in the community. For RHBQ the support cycles and windows are longer - hence 2.7.x still maintained.

I realize we could use an update to the documentation of update policies thus added that to my list of TODO's - that should help you get the right endoflife.date updates.

**Reply by marcwrobel** [reply]
Thank you for your prompt reply to my questions @maxandersen. I did not know there was Red Hat builds of Quarkus before yesterday and you helped me to understand what it is.

From what I can see on the [RHBQ product page](https://access.redhat.com/products/quarkus) you need a subscription to get support from Red Hat, which is normal. But I created an application using https://code.quarkus.redhat.com/ and did not see anything restricting usage of RHBQ. Is it possible to use freely RHBQ in production and for commercial use without a subscription (i.e. without the Red Hat support) ?

I did not find this information on https://access.redhat.com/products/quarkus (especially in the _Support Program & Policies_ section). I saw those links (https://access.redhat.com/discussions/5585571, https://unix.stackexchange.com/q/556289) but the discussions are about RHEL (and are still not clear to me) so I do not know if it applies to RHBQ.

**Comment by tqvarnst** [comment]
Hi @marcwrobel! 
AFAIK RHBQ is free for development purposes but requires a subscription to run in production. Also, I wouldn't recommend using RHBQ without a subscription. The reason is that without a subscription you are not guaranteed to get fixes implemented in a product release stream.

If you absolutely do not want to use a subscription we are investigating adding LTS releases also to the community, we plan to announce something at the end of the year. 

One more thing, if you have any other Red Hat products, like RHEL, OpenShift, etc you are likely already entitled to an RHBQ subscription.

I hope this helps
Thomas Qvarnström, Product Manger, Red Hat

**Reply by marcwrobel** [reply]
Thanks a lot for this clarification @tqvarnst, it helps a lot.

**Comment by adriens** [comment]
[Quarkus endoflife.date page](https://endoflife.date/quarkus-framework) is now much cleaner, thanks a lot guys for great and efficeient collab' :muscle: 

![image](https://user-images.githubusercontent.com/5235127/201495784-004b595f-945c-4fd0-9316-42ad9aef49bd.png)
...
![image](https://user-images.githubusercontent.com/5235127/201495791-44ae6f34-b913-4302-940f-a2d50dd728c5.png)

**Comment by adriens** [comment]
![image](https://user-images.githubusercontent.com/5235127/201495809-ffcb0f47-a604-48fa-bd33-50a2d441abcb.png)
